### ISSUE TITLE
When upgrading to 24.1.0 under "beta-private-cluster-update-variant" node_pools are being recreated

### ISSUE THREAD

**Issue by skakauridze-clgx** [terraform-google-modules/terraform-google-kubernetes-engine#1558]
### TL;DR

When upgrading to 24.1.0 under "beta-private-cluster-update-variant" node_pools are being recreated.

Because this module doesn't support 0 downtime node_pool recreation and is not cordoning and draining old node_pool, we don't want to trigger a node_pool recreation:
![nodepool recreation](https://user-images.githubusercontent.com/47898136/217080071-1eb0653e-8439-4ca7-a6a4-5f213172480f.png)

To avoid this behavior form the screeshot, we had to revert node_pool recreation resoures to its previous value before version 24.1.0:
![Screen Shot 2023-02-06 at 12 41 16 PM](https://user-images.githubusercontent.com/47898136/217080486-c226b87c-12f8-4b84-901f-af753ba49cb1.png)

What would be your suggestion? How should we handle this situation the most efficiently?

### Terraform Configuration

```hcl
This was provided in a screenthos
```

### Terraform Version

```sh
1.0.8
```

**Comment by cheddarwhizzy** [comment]
I had a similar situation where a change to the node pools wanted to force recreate all of them causing downtime in the cluster. I ended up making a separate "rolling update" step in our CI/CD process to check the plan file for new pools, recreated pools, and deleted pools, Then I run `gcloud container clusters update` and `gcloud container clusters resize` to disable autoscaling on the pool and resize it to 0 hosts (this runs a drain on the hosts, respecting PDBs). Then apply -target=<node_pool_address> for each one and waiting for the cluster's pods to all be ready. Kinda hacky - yes. I came to the issue queue to see if blue/green was added yet and was hoping that would resolve needing to do this. Idk why EKS and GKE don't follow this "drain one node at a time" like Kops does. 

```
# new pools
terragrunt show -json plan | jq -r '.output_changes.node_pools_names.after-.output_changes.node_pools_names.before

# replaced pools
terragrunt show -json plan | jq -r '.resource_changes[] | select(.change.replace_paths != null).address'

# deleted pools
terragrunt show -json plan | jq -r '.output_changes.node_pools_names.before-.output_changes.node_pools_names.after

```
